Documentation updates for 0.16 (#864)
reasonml/reason-react: docs/usereducer-hook.md

---
title: useReducer
---

[React.js docs for useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer)

> useReducer is usually preferable to useState when you have complex state logic that involves multiple sub-values or when the next state depends on the previous one. useReducer also lets you optimize performance for components that trigger deep updates because you can pass dispatch down instead of callbacks.

```reason

> /* we create a type for the action, but action can be anything */

Reviewer: ```suggestion
/* we can create anything as the type for action, here we use a variant with 2 cases. */
```

type action =
  | Increment
  | Decrement;

> /* similarly on 'state', it can be anything. In this case, it's an int */

Reviewer: ```suggestion
/* `state` could also be anything. In this case, we want an int */
```
